polaris-core/src/main/java/org/apache/polaris/core/persistence/PolarisCredentialsBootstrap.java

```diff
@@ -63,6 +66,39 @@ public static PolarisCredentialsBootstrap fromString(@Nullable String credential
         : EMPTY;
   }
 
+  /**
+   * Parse a JSON array of credentials. Example: """ [ {"realm": "a", "principal": "root",
+   * "clientId": "abc123", "clientSecret": "xyz987"}, {"realm": "b", "principal": "boot",
+   * "clientId": "boot-id", "clientSecret": "boot-secret"}, ] """
+   */
+  public static PolarisCredentialsBootstrap fromJson(String json) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    try {
+      List<Map<String, String>> credentialsList =
+          objectMapper.readValue(json, new TypeReference<List<Map<String, String>>>() {});
+      List<String> formattedList = new ArrayList<>();
+      for (Map<String, String> entry : credentialsList) {
+        String realm = entry.get("realm");
+        String principal = entry.get("principal");
+        String clientId = entry.get("clientId");
+        String clientSecret = entry.get("clientSecret");
+        if (realm == null || principal == null || clientId == null || clientSecret == null) {
+          throw new IllegalArgumentException("Invalid JSON format for credentials: " + json);
+        }
+        if (!principal.equals(PolarisEntityConstants.ROOT_PRINCIPAL_NAME)) {
+          throw new IllegalArgumentException(
+              String.format(
+                  "Invalid principal %s. Expected %s.",
+                  principal, PolarisEntityConstants.ROOT_PRINCIPAL_NAME));
+        }
+        formattedList.add(String.join(",", realm, principal, clientId, clientSecret));
+      }
+      return fromList(formattedList);
```

Comment:
Re-parsing from a generated list looks too complex to me. Why not create a `PolarisCredentialsBootstrap` object directly?